### PR TITLE
Text indent, production template example

### PR DESCRIPTION
**Pull request by AndreaGreco**
Add some function for improve text indention, add a production template example.

set gin in debug.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,20 +13,23 @@ make
 
 ## Usage
 
-1. Specify telegram token in ```config.yaml```:
+1. Create Telegram bot with BotFader, it will return you bot token
+
+2. Add your bot to a group. It should report group id now, here you find ```CHAT-ID```
```
Comment (hryamzik) [inline README.md]: That's not a requirement, chats work fine as well.

File: README.md [hunk 2]
```diff
@@ -82,4 +88,37 @@ Best way for build your custom template is:
 
 ```-d``` options will enable ```debug``` mode and template file will reload every message, else template is load once on startup.
 
-Is provided as [default template file](testdata/default.tmpl) with all possibile variable. Remember that telegram bot support HTML check [here](https://core.telegram.org/bots/api#html-style) list of aviable tags.
```
Comment (hryamzik) [inline README.md]: Why did you remove link to a template?

**Comment by hryamzik** [comment]
Hi @AndreaGreco, some tests are failing:

```
ok 1 - emptyValue.json template none
ok 2 - noGenURL.json template none
ok 3 - simpe.json template none
ok 4 - production_example.json template none
not ok 5 - emptyValue.json template default.tmpl
not ok 6 - noGenURL.json template default.tmpl
not ok 7 - simpe.json template default.tmpl
not ok 8 - production_example.json template default.tmpl
not ok 9 - emptyValue.json template detailed_vars.tmpl
not ok 10 - noGenURL.json template detailed_vars.tmpl
not ok 11 - simpe.json template detailed_vars.tmpl
not ok 12 - production_example.json template detailed_vars.tmpl
not ok 13 - emptyValue.json template production_example.tmpl
not ok 14 - noGenURL.json template production_example.tmpl
not ok 15 - simpe.json template production_example.tmpl
not ok 16 - production_example.json template production_example.tmpl
Failed 12/16 subtests 

Test Summary Report
-------------------
t/curl.t (Wstat: 0 Tests: 16 Failed: 12)
  Failed tests:  5-16
Files=1, Tests=16, 13 wallclock secs ( 0.04 usr  0.00 sys +  0.29 cusr  0.02 csys =  0.35 CPU)
Result: FAIL
```

Didn't figure out what's wrong yet. Please see my review comments.

**Comment by hryamzik** [comment]
Here's how tests fail:

```
****** Run prometheus_bot with template testdata/default.tmpl ******
Load template file:testdata/default.tmpl
You must define time_zone of your bot
****** Run prometheus_bot with template testdata/detailed_vars.tmpl ******
Load template file:testdata/detailed_vars.tmpl
You must define time_zone of your bot
****** Run prometheus_bot with template testdata/production_example.tmpl ******
Load template file:testdata/production_example.tmpl
You must define time_zone of your bot
```

But `config.yaml` in test environment is fine:
```yml
telegram_token: "***"
time_zone: "Europe/Rome"
```

**Comment by hryamzik** [comment]
@AndreaGreco can I also ask you to run `go fmt` on your code? There're some formatting issues that could be easily avoided with running go formatter.

**Comment by hryamzik** [comment]
Integration tests run fine on current master so it's definitely an issue with your code.

File: README.md (hunk 1 above)
Comment (AndreaGreco) [inline README.md]: Ok, i'm going to remove about group, in doc.

Comment (hryamzik) [inline README.md]: I'd leave it as an advise, not requirement.

File: README.md (hunk 2 above)
Comment (AndreaGreco) [inline README.md]: This will restored

File: main.go [hunk 3]
```diff
@@ -144,7 +260,9 @@ func main() {
 		*debug = false
 		tmpH = nil
 	}
-	//bot.Debug = true
+    if !(*debug) {
```
Comment (hryamzik) [inline main.go]: Here's an issue with code formatting ;-)

Comment (AndreaGreco) [inline main.go]: now i run fmtgo -w in last commit before send you a fixed pull request.
i dident know of it existence before you talking here

Comment (hryamzik) [inline main.go]: Some text editors do this for you. ;-)

**Comment by hryamzik** [comment]
@AndreaGreco thanks! Can I ask you to verify that tests do not fail? There're some issues with CI so we won't see them here in PR status but they still do fail.

**Comment by AndreaGreco** [comment]
My test is all passed

```
go build -o prometheus_bot
prove -v
t/curl.t .. 
1..16
ok 1 - simpe.json template none
ok 2 - noGenURL.json template none
ok 3 - production_example.json template none
ok 4 - emptyValue.json template none
ok 5 - simpe.json template detailed_vars.tmpl
ok 6 - noGenURL.json template detailed_vars.tmpl
ok 7 - production_example.json template detailed_vars.tmpl
ok 8 - emptyValue.json template detailed_vars.tmpl
ok 9 - simpe.json template default.tmpl
ok 10 - noGenURL.json template default.tmpl
ok 11 - production_example.json template default.tmpl
ok 12 - emptyValue.json template default.tmpl
ok 13 - simpe.json template production_example.tmpl
ok 14 - noGenURL.json template production_example.tmpl
ok 15 - production_example.json template production_example.tmpl
ok 16 - emptyValue.json template production_example.tmpl
ok
All tests successful.
Files=1, Tests=16, 23 wallclock secs ( 0.02 usr  0.00 sys +  0.34 cusr  0.17 csys =  0.53 CPU)
Result: PASS
```

**Comment by hryamzik** [comment]
I've pulled your branch and mine fail:

```
$ prove -v
t/curl.t .. 
1..16
ok 1 - emptyValue.json template none
ok 2 - noGenURL.json template none
ok 3 - production_example.json template none
ok 4 - simpe.json template none
not ok 5 - emptyValue.json template default.tmpl
not ok 6 - noGenURL.json template default.tmpl
not ok 7 - production_example.json template default.tmpl
not ok 8 - simpe.json template default.tmpl
not ok 9 - emptyValue.json template detailed_vars.tmpl
not ok 10 - noGenURL.json template detailed_vars.tmpl
not ok 11 - production_example.json template detailed_vars.tmpl
not ok 12 - simpe.json template detailed_vars.tmpl
not ok 13 - emptyValue.json template production_example.tmpl
not ok 14 - noGenURL.json template production_example.tmpl
not ok 15 - production_example.json template production_example.tmpl
not ok 16 - simpe.json template production_example.tmpl
Failed 12/16 subtests 

Test Summary Report
-------------------
t/curl.t (Wstat: 0 Tests: 16 Failed: 12)
  Failed tests:  5-16
Files=1, Tests=16, 12 wallclock secs ( 0.02 usr  0.00 sys +  0.41 cusr  0.25 csys =  0.68 CPU)
Result: FAIL
```

In fact it just doesn't start:

```
./prometheus_bot -d -t testdata/default.tmpl 
Load template file:testdata/default.tmpl
You must define time_zone of your bot
```

Time zone is specified in config file:

```
$ cat config.yaml|sed -r 's/(telegram_token: ).*/\1"sensored"/'
telegram_token: "sensored"
time_zone: "Europe/Rome"
```

**Comment by AndreaGreco** [comment]
Yes fix config, miss a paraneter

**Comment by hryamzik** [comment]
OK, I've found the issue:

```yml
type Config struct {
	TelegramToken string `yaml:"telegram_token"`
	TemplatePath  string `yaml:"template_path"`
	TimeZone      string `yaml:"template_time_zone"`
	TimeOutFormat string `yaml:"template_time_outdata"`
}
```

Let's not change option names, that's going to break existing installations.

**Comment by AndreaGreco** [comment]
Ok, fixed i think

**Comment by hryamzik** [comment]
Tests pass now, thanks!

**Comment by AndreaGreco** [comment]
To you sorry, but was difficult merge, i do this work lot't time ago 2 month.
Never fired pull request
